### PR TITLE
PEN-1736 SVG Icon updates to provide context prop

### PR DESCRIPTION
**Pull request by matthewroach**
## Description

Update icon components to provide a context prop, to allow for the consumer to decide what context the icon will be used in. Thus allowing the SVG to have the correct attributes for accessibility.

## Jira Ticket
- [PEN-1736](https://arcpublishing.atlassian.net/browse/PEN-1736)

## Test Steps

- Verify icon components still render as before
- Verify icon components default to presentational and include the correct attributes
- Verify if an icon is used in the `image` context the correct attributes are applied

## Dependencies or Side Effects
_Examples of dependencies or side effects are:_
- Additional settings that will be required in the blocks.json
- Changes to the custom fields which will require users to reconfigure features
- Update to css framework or SDK
- Dependency on another PR that needs to be merged first

## Review Checklist
_The author of the PR should fill out the following sections to ensure this PR is ready for review._
- [x] Confirmed all the test steps above are working
- [x] Confirmed there are no linter errors
- [x] Confirmed this PR has reasonable code coverage
  - [x] Confirmed this PR has unit test files
  - [x] Ran `npm test`, made sure all tests are passing
  - [x] If the amount of work to write unit tests for this change are excessive,
please explain why (so that we can fix it whenever it gets refactored).
- [x] Confirmed relevant documentation has been updated/added.

### CODE REVIEW

File: src/components/icons/ArrowRightIcon.tsx [hunk 1]
```diff
@@ -1,11 +1,12 @@
 import React from 'react';
+import additionalSVGProps from './Helpers';
 
 type IconProps = import('./types').default;
 
 const ArrowRightIcon: React.FC<IconProps> = ({
-  width = 24, height = 24, fill = '#000', title = 'Arrow Right', description = '',
+  width = 24, height = 24, fill = '#000', title = 'Arrow Left', description = '', context = 'presentational',
```
Comment (vinnie-olsen) [inline src/components/icons/ArrowRightIcon.tsx]: `title` should be `Arrow Right` here

File: src/components/icons/ArrowUpIcon.tsx [hunk 2]
```diff
@@ -1,11 +1,12 @@
 import React from 'react';
+import additionalSVGProps from './Helpers';
 
 type IconProps = import('./types').default;
 
 const ArrowUpIcon: React.FC<IconProps> = ({
-  width = 24, height = 24, fill = '#000', title = 'Arrow Up', description = '',
+  width = 24, height = 24, fill = '#000', title = 'Arrow Left', description = '', context = 'presentational',
```
Comment (vinnie-olsen) [inline src/components/icons/ArrowUpIcon.tsx]: `title` should be `Arrow Up` here

File: src/components/icons/EllipsisHorizontalIcon.tsx [hunk 3]
```diff
@@ -1,11 +1,12 @@
 import React from 'react';
+import additionalSVGProps from './Helpers';
 
 type IconProps = import('./types').default;
 
 const EllipsisHorizontalIcon: React.FC<IconProps> = ({
-  width = 24, height = 24, fill = '#000', title = 'EllipsisHorizontal', description = '',
+  width = 24, height = 24, fill = '#000', title = 'EllipsisHorizontal', description = '', context = 'presentational',
```
Comment (vinnie-olsen) [inline src/components/icons/EllipsisHorizontalIcon.tsx]: Should this be changed to a `title` of `Ellipsis Horizontal` to be consistent in format with the `Ellipsis Vertical` title below?

**Review by vinnie-olsen — CHANGES_REQUESTED** [review]

**Comment by vinnie-olsen** [comment]
@matthewroach one other thing I just thought of. On https://arcpublishing.atlassian.net/browse/PEN-676 , Brent had mentioned removing the `<desc>` tag for non-image SVG's and we discussed deferring that until a later ticket. Was this the ticket that was intended to remove those tags? I am still seeing those `<desc>` tags render on non-image SVG's but other than that, everything else looks good!

**Comment by matthewroach** [comment]
@vinnie-olsen The `<desc>` children is a valid SVG item, that does have a use. For non-image SVG's the title and desc elements will now be ignored due to the use of `aria-hidden` - I've updated the code to not render those elements in non-image context, this helps with performance and redundant code

File: .eslintrc.js [hunk 4]
```diff
@@ -17,7 +17,8 @@ module.exports = {
     'react/prop-types': 'off',
     'react/require-default-props': 'off',
     'react/static-property-placement': 'off',
-    'import/extensions': 'off'
+    'import/extensions': 'off',
+    'react/jsx-props-no-spreading': 'off',
```
Comment (JackHowa) [inline .eslintrc.js:21]: thanks

**Review by JackHowa — APPROVED** [review]
storybook looks good, thanks

**Pull request merged**